### PR TITLE
chore: move whitelist to data.ts

### PR DESCRIPTION
**Pull request by thelostone-mc**
## Description

- reduces calls to whitelist API
- ensure relevant grants are shown on the grants landing page
- moves the filter function to getAllGrants

Testing 

- ✅ Homepage shows filtered grants
- ✅Grants list shows filtered grants
- ✅ Next / Prev also shows filtered grants
- ✅ cart shows approved grants
- ✅ ~Access unapproved grants via URL  http://localhost:3000/#/dgrants/1?unverified=true~

closes https://github.com/dcgtc/dgrants/issues/451
closes https://github.com/dcgtc/dgrants/issues/407
closes https://github.com/dcgtc/dgrants/issues/429

~Q: Another way to go about this is to have a separate function in data.ts which just fetches the whitelist and have the frontend do the filtering
Q: As opposed to having data return grants. Would it make sense to return whitelistGrants and allGrants ?~
~What's missing ? Since whitelist is done at grant data layer you cannot access the grant URL directly.
Need to sync up with team if it makes sense a new route URL which would allow viewing unapproved grants by URL~

### CODE REVIEW

File: app/src/views/GrantRegistryGrantDetail.vue [hunk 1]
```diff
@@ -324,10 +322,13 @@ function useGrantDetail() {
...
   const route = useRoute();
 
+  // --- show unverified grants if unverified = true
+  const grants = route.query.unverified ? allGrants : approvedGrants;
+
   // --- expose grant data ---
   const grant = computed(() => {
     if (!grants.value) return null; // array type unsupported
-    return grants.value[grantId.value];
+    return grants.value.filter((grant) => grant.id.toString() == route.params.id)[0];
```
Comment (mds1) [inline app/src/views/GrantRegistryGrantDetail.vue:331]: Why was this changed? The original approach was O(1) and now this is O(n), so it might be slow if there's a lot of grants (though realistically the speed change will probably be unnoticeable)

File: app/src/components/BaseHeader.vue [hunk 2]
```diff
@@ -28,15 +28,15 @@
     <span>
       <!-- optional navigation to jump to next item in a list -->
       <router-link
-        v-if="nextPath"
+        v-if="nextPath && !isUnverified"
```
Comment (mds1) [inline app/src/components/BaseHeader.vue:31]: If I build for polygon and directly navigate to `http://localhost:3000/#/dgrants/19` (an unverified grant) I see a page that just says "No grant selected". If I navigate to `http://localhost:3000/#/dgrants/19?unverified=true`, then the page loads. 

It seems a bit odd that you'd need to add that flag just to view a grant, so just want to confirm that's the desired behavior

Comment (mk1r) [reply app/src/components/BaseHeader.vue:31]: I think this is not desired - tested this is well with grant 9. On main, if we navigate to a non-whitelisted grant like 9 or 19 for example, the grant shows is still viewable which is what we want.

File: app/src/views/GrantRegistryGrantDetail.vue (hunk 1 above)
Comment (mk1r) [reply app/src/views/GrantRegistryGrantDetail.vue:331]: just looked into it, and I believe it's because `grantId.value` is the index value for non-whitelisted grants, but since we're fetching the specific grant from whitelisted grants, there is an index mismatch and it needs to pull the id instead so it's filtering for that

there probably is a more efficient way to handle this though, and this might be related to the other comment about making non-whitelisted grants viewable / unviewable

Comment (mds1) [reply app/src/views/GrantRegistryGrantDetail.vue:331]: Ah, that makes total sense, thanks!

File: app/src/components/BaseHeader.vue (hunk 2 above)
Comment (thelostone-mc) [reply app/src/components/BaseHeader.vue:31]: @mk1r  + @mds1 pushed in a fix. could you recheck this now
Simplified it a bit

**Review by mk1r — APPROVED** [review]
looking good! nice changes

**Comment by mk1r** [comment]
oh but actually, don't we want to show non-whitelisted grants as well? @phutchins can also provide some clarity

thought that was changed but double checked and perhaps not?

File: app/src/store/data.ts [hunk 3]
```diff
@@ -112,6 +114,9 @@ export default function useDataStore() {
       return grants;
     }, {});
 
+
+    const approvedGrantsList = await getApprovedGrants(grantsData.grants);
```
Comment (paffur) [inline app/src/store/data.ts:118]: https://github.com/dcgtc/dgrants.wiki.git

**Review by paffur — APPROVED** [review]

**Review by mds1 — APPROVED** [review]
